x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java

```diff
@@ -441,10 +443,12 @@ public XPackClientPlugin(final Settings settings) {
                 new NamedWriteableRegistry.Entry(EvaluationMetricResult.class, ScoreByThresholdResult.NAME, ScoreByThresholdResult::new),
                 new NamedWriteableRegistry.Entry(EvaluationMetricResult.class, ConfusionMatrix.NAME.getPreferredName(),
                         ConfusionMatrix.Result::new),
-                // ML - Inference
+                // ML - Inference preprocessing
                 new NamedWriteableRegistry.Entry(PreProcessor.class, FrequencyEncoding.NAME.getPreferredName(), FrequencyEncoding::new),
                 new NamedWriteableRegistry.Entry(PreProcessor.class, OneHotEncoding.NAME.getPreferredName(), OneHotEncoding::new),
                 new NamedWriteableRegistry.Entry(PreProcessor.class, TargetMeanEncoding.NAME.getPreferredName(), TargetMeanEncoding::new),
+                // ML - Inference models
```

Comment:
I tried to avoid using "models" since there is a long tradition of overloading this term. In JSON schema the section is called `evaluation`. I am not particularly invested in the term, but we should stick to the same terminology everywhere.

This also may mean that we rename "evaluation" to "model" in the JSON schema.

Reply:
I really like using `model` as it is a unified name with the `ensemble` and its nested `models` object.

It seems logical to me that a `model` that is an `ensemble` will have many `models`. If we choose to use `evaluation` I think `ensemble` should have an `evaluations` field.